test/crypto-service/test_accountUpdateTransaction.js

```diff
@@ -0,0 +1,1041 @@
+      verifyAccountKeyUpdate(String(ecdsaSecp256k1PublicKey.key).substring(28).toLowerCase());
+    });
+
+    it("(#3) Updates the key of an account to a new valid ED25519 private key", async function () {
+      // Generate a new ED25519 private key for the account.
+      const ed25519PrivateKey = await JSONRPCRequest("generateKey", {
+        type: "ed25519PrivateKey"
+      });
+      if (ed25519PrivateKey.status === "NOT_IMPLEMENTED") this.skip();
+
+      // Generate the corresponding ED25519 public key.
+      const ed25519PublicKey = await JSONRPCRequest("generateKey", {
+        type: "ed25519PublicKey",
+        fromKey: ed25519PrivateKey.key
+      });
+      if (ed25519PublicKey.status === "NOT_IMPLEMENTED") this.skip();
+
+      // Attempt to update the key of the account with the new ED25519 private key.
+      const response = await JSONRPCRequest("updateAccount", {
+        accountId: accountId,
+        key: ed25519PrivateKey.key,
+        commonTransactionParams: {
+          signers: [
+            accountPrivateKey,
+            ed25519PrivateKey.key
+          ]
+        }
+      })
+      if (response.status === "NOT_IMPLEMENTED") this.skip();
+
+      // Verify the account key was updated (use raw key for comparison, ED25519 public key DER-encoding has a 12 byte prefix).
+      verifyAccountKeyUpdate(String(ed25519PublicKey.key).substring(24).toLowerCase());
+    });
+
+    it("(#4) Updates the key of an account to a new valid ECDSAsecp256k1 private key", async function () {
+      // Generate a new ECDSAsecp256k1 private key for the account.
+      const ecdsaSecp256k1PrivateKey = await JSONRPCRequest("generateKey", {
+        type: "ecdsaSecp256k1PrivateKey"
+      });
+      if (ecdsaSecp256k1PrivateKey.status === "NOT_IMPLEMENTED") this.skip();
+
+      // Generate the corresponding ECDSAsecp256k1 public key.
+      const ecdsaSecp256k1PublicKey = await JSONRPCRequest("generateKey", {
+        type: "ecdsaSecp256k1PublicKey",
+        fromKey: ecdsaSecp256k1PrivateKey.key
+      });
+      if (ecdsaSecp256k1PublicKey.status === "NOT_IMPLEMENTED") this.skip();
+
+      // Attempt to update the key of the account with the new ECDSAsecp256k1 public key.
+      const response = await JSONRPCRequest("updateAccount", {
+        accountId: accountId,
+        key: ecdsaSecp256k1PrivateKey.key,
+        commonTransactionParams: {
+          signers: [
+            accountPrivateKey,
+            ecdsaSecp256k1PrivateKey.key
+          ]
+        }
+      })
+      if (response.status === "NOT_IMPLEMENTED") this.skip();
+
+      // Verify the account key was updated (use raw key for comparison, compressed ECDSAsecp256k1 public key DER-encoding has a 14 byte prefix).
+      verifyAccountKeyUpdate(String(ecdsaSecp256k1PublicKey.key).substring(28).toLowerCase());
+    });
+
+    it("(#5) Updates the key of an account to a new valid KeyList of ED25519 and ECDSAsecp256k1 private and public keys", async function () {
```

Comment:
it says that the keylist contains the 2 types of keys, but this is not true.

Reply:
Added